src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -229,10 +230,27 @@ private String href(String path, boolean isDir) {
         return href;
     }
 
-    public String formatHtmlLink(String link) {
+    private static String formatHtmlLink(String link) {
         return link.replaceAll("&", "&amp;");
     }
 
+    private static String formatPathURL(String input) {
```

Comment:
@NielsCharlier while this is nice and fast, you have pretty much reimplemented URLEncoder.encode. For maintainability and consistency, please implement this by splitting the path on "/" and then encoding with URLEncoder.encode to UTF-8. This implementation misses several characters like round and square brackets. Why not let  URLEncoder.encode do the work.

Also, as discussed on the mailing list, ResponseUtils.buildURL is sorely in need of this code, so if you use ResponseUtils.buildURL everywhere and fix its implementation to percent encode path elements, you will fix two problems at once.